test/index.test.js

```diff
@@ -1,41 +1,218 @@
 'use strict';
 
-const expect = require('chai').expect;
-const Hapi   = require('hapi');
+const Bluebird = require('bluebird');
+const expect   = require('chai').expect;
+const Hapi     = require('hapi');
+const Redis    = require('then-redis');
+
+const redisClient = Redis.createClient({
+  port: '6379',
+  host: 'localhost',
+  enable_offline_queue: false
```

Comment:
>By default, if there is no active connection to the Redis server, commands are added to a queue and are executed once the connection has been established. Setting enable_offline_queue to false will disable this feature and the callback will be executed immediately with an error, or an error will be emitted if no callback is specified.

From https://github.com/NodeRedis/node_redis#options-object-properties

I think queuing system is useful to have and is much better than just throwing errors.